### ISSUE TITLE
minAppVersion warning, but using `requireApiVersion()` as a gate

### ISSUE THREAD

**Issue by chrisgurney** [obsidianmd/eslint-plugin#151]
I'm getting this warning:

> 'Plugin.registerCliHandler' requires Obsidian v1.12.2, but minAppVersion is 1.11.0.

However, I have an `requireApiVersion('1.12.2')` in my code path (up one class) before the handler is registered.

```
if (Platform.isDesktop && requireApiVersion('1.12.2')) this.cli.register();
```

And here's `this.cli.register()`:
```
    register(): void {
...
        // register the commands with Obsidian
        for (const cmd of this.cliDefinition.commands) {
            this.ntb.registerCliHandler(cmd.id, cmd.description, cmd.flags, this.cliHandlers.get(cmd.id));
        }

    }
```

**Comment by chrisgurney** [comment]
Workaround from Johan [on Discord](https://discord.com/channels/686053708261228577/840286264964022302/1517038528637173761):

> You just need to wrap all 1.13 functions in api checks. Then it works. Cannot do a branch check etc.
